packages/cli/src/util/use-adaptors-repo.ts

```diff
@@ -0,0 +1,52 @@
+import { readFile } from 'node:fs/promises';
+import path from 'node:path';
+import assert from 'node:assert';
+
+import { Logger } from '@openfn/logger';
+import { getNameAndVersion } from '@openfn/runtime';
+
+export const validateMonoRepo = async (repoPath: string, log: Logger) => {
+  try {
+    const raw = await readFile(`${repoPath}/package.json`, 'utf8');
+    const pkg = JSON.parse(raw);
+    assert(pkg.name === 'adaptors');
+  } catch (e) {
+    log.error(`ERROR: Monorepo not found at ${repoPath}`);
+    throw new Error('Monorepo not found');
+  }
+};
+
+// Convert an adaptor name into a path to the adaptor in the monorepo
+export const updatePath = (adaptor: string, repoPath: string, log: Logger) => {
+  if (adaptor.match('=')) {
+    // Should do nothing if a path is already provided
+    return adaptor;
+  }
+
+  const { name, version } = getNameAndVersion(adaptor);
+  if (version) {
+    // version numbers are ignored with warning
+    log.warn(
+      `Warning: Ignoring version specifier on ${adaptor} as loading from the adaptors monorepo`
```

Comment:
Actually, if you ask for a specific version, probably we should use that version and ignore the monorepo. Like `openfn execute -a http@1.0.0` should lookup the 1.0.0 version in the repo, not whatever is in the monorepo.